zerver/lib/message.py

```diff
@@ -847,18 +851,28 @@ def apply_unread_message_event(user_profile: UserProfile,
     if 'mentioned' in flags:
         state['mentions'].add(message_id)
 
-# 4 hours is probably frequent enough to encourage upgrade
-# and largely mitigates the cache-miss problem. We also
-# don't need to restrict messages to exactly
-# message_visibility_limit. The query is also reasonably fast
-# so cache misses can be managed without cron job.
-@cache_with_key(realm_first_visible_message_id_cache_key, timeout=3600*4)
-def get_first_visible_message_id(realm: Realm) -> int:
-    if realm.message_visibility_limit is None:
-        return 0
+def estimate_recent_messages(realm: Realm, hours: int) -> int:
+    stat = COUNT_STATS['messages_sent:is_bot:hour']
+    d = timezone_now() - datetime.timedelta(seconds=hours * 60)
```

Comment:
Yikes! Glad I caught this -- this is why we always put units into the identifier names, so this kind of thing is possible to catch.
> seconds=hours * 60

There are 3600 seconds in an hour, not 60. :clock10: